### PR TITLE
Appshell item template

### PR DESCRIPTION
**Pull request by licon4812**
### Description of Change

This PR adds AppShell item templates so that users can add an Appshell to their application if they started with a blank project

### CODE REVIEW

File: src/Templates/src/templates/maui-appshell-csharp/NewAppshell1.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+namespace MauiApp1;
+
+public class NewAppshell1 : Shell
+{
+	public NewAppshell1()
+	{
+		shell = new Shell()
```
Comment (Eilon) [inline src/Templates/src/templates/maui-appshell-csharp/NewAppshell1.cs:7]: I think this field is not declared anywhere?

File: src/Templates/src/templates/maui-appshell-xaml/NewAppshell1.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿namespace MauiApp1;
+
+public partial class NewAppshell1 : Shell
+{
+    public NewAppshell1()
+    {
+		InitializeComponent();
```
Comment (Eilon) [inline src/Templates/src/templates/maui-appshell-xaml/NewAppshell1.xaml.cs:7]: Formatting here is mixing tabs and spaces.

File: src/Templates/src/Microsoft.Maui.Templates.csproj [hunk 3]
```diff
@@ -58,27 +58,16 @@
     <BeforePack>_UpdateTemplateVersions</BeforePack>
   </PropertyGroup>
```
Comment (Eilon) [inline src/Templates/src/Microsoft.Maui.Templates.csproj:60]: Were any changes to this file deliberate? If not, please undo the changes to this file.

File: src/Templates/src/templates/maui-appshell-csharp/NewAppshell1.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+		{
+			Children =
+			{
+				new FlyoutItem{ Title = "Cats" }
```
Comment (Eilon) [inline src/Templates/src/templates/maui-appshell-csharp/NewAppshell1.cs:11]: I suggest instead of cats/dogs, to use a more generic `Item 1` and `Item 2`

**Review by Eilon — CHANGES_REQUESTED** [review]
Hi, I left a few comments in this PR, but we also need to verify that this feature aligns with scenarios that we want to support. Tagging @PureWeen to confirm that.

**Comment by PureWeen** [comment]
@jfversluis @davidortinau  thoughts?

File: src/Templates/src/templates/maui-appshell-csharp/NewAppshell1.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
+namespace MauiApp1;
+
+public class NewAppshell1 : Shell
+{
+	public NewAppshell1()
+	{
+		var shell = new Shell()
+		{
+			Children =
```
Comment (davidortinau) [inline src/Templates/src/templates/maui-appshell-csharp/NewAppshell1.cs:9]: Same comment as the XAML version, we shouldn't provide code that doesn't work as expected, but instead a sample of code that would work when implemented specific to the app. 

Also this isn't a valid implementation. It should look something like this:

```csharp
public class MyShell : Shell
{
	public MyShell()
	{
		this.Items.Add(new FlyoutItem
		{
			Title = "Dashboard",
			Items =
			{
				new ShellContent
				{
					ContentTemplate = new DataTemplate(typeof(DashboardPage)),
				}
			}
		});

		this.Items.Add(new FlyoutItem
		{
			Title = "Add Vocab",
			Items =
			{
				new ShellContent
				{
					ContentTemplate = new DataTemplate(typeof(AddVocabularyPage)),
				}
			}
		});
	}
}
```

**Comment by davidortinau** [comment]
Generally I'm okay with having item templates for this, however we should consider if we want to promote the notion of using multiple Shells in a single project. What we have been seeing is devs using multiple Shells for scenarios that are better achieved with the features of a single Shell using the visibility and enabled properties. For example, swapping shell implementations on MainPage is not performant compared to using those properties.

**Comment by licon4812** [comment]
Maybe have the appshell as an option in the project creation instead of an item template?

**Comment by jfversluis** [comment]
Let's take a step back: what is the thing we're trying to solve with this? What triggered you @licon4812 to implement this?

I don't think we ever heard much ask for adding this by other users? Granted, this will be something that is extra and will not bother people unless they go looking for it. On the other hand, we will be the ones having to maintain this and once it's in there, its hard to remove it again.

**Comment by licon4812** [comment]
Basically, I just want an option to add a shell to a project IF the project was created without one. Basically just a minor thing

**Comment by licon4812** [comment]
@jfversluis  that is a very good point, I was basically just wanting a time savour, which might already exist in plugins anyway.

**Comment by jfversluis** [comment]
All things considered I don't think this is something we would want to do right now. But again: your efforts and point of view are appreciated. Looking forward to your next contribution!

**Pull request closed without merging**